org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/settings/XMLFileAssociation.java

```diff
@@ -49,32 +48,11 @@ public void setSystemId(String systemId) {
 	}
 
 	public boolean matches(String uri) {
-		try {
-			return matches(new URI(uri));
-		} catch (Exception e) {
-			return false;
-		}
+		return URIUtils.matches(pattern, uri, pathMatcher);

```

Comment:
In your case pathMatcher is every time null. In the original XMLFileAssociation, the pathMatcher is null the first call of matches and initialized and the second time of call of matches it is initialized.

I think we should not have URIUtils.matches, but a base class for path matcher (like PatchMatcher class). XMLFileAssociation could extend PatchMatcher  and your XMLExcludedSymbolFile too